### PR TITLE
Use the firrtl-test.jar & delete the duplicated sources

### PR DESCRIPTION
**Pull request by davidbiancolin**
See ucb-bar/chipyard#551.

### CODE REVIEW

File: sim/Makefile [hunk 1]
```diff
@@ -42,12 +42,22 @@ SBT ?= java $(JAVA_ARGS) -jar $(rocketchip_dir)/sbt-launch.jar
 # Manage the FIRRTL dependency manually
 FIRRTL_SUBMODULE_DIR ?= $(chipyard_dir)/tools/firrtl
 FIRRTL_JAR ?= $(chipyard_dir)/lib/firrtl.jar
-$(FIRRTL_JAR): $(shell find $(FIRRTL_SUBMODULE_DIR)/src/main/scala -iname "*.scala")
+FIRRTL_TEST_JAR ?= $(chipyard_dir)/test_lib/firrtl.jar
+
+firrtl_srcs := $(shell find $(FIRRTL_SUBMODULE_DIR) -iname "*.scala")
+
+$(FIRRTL_JAR): $(firrtl_srcs)
```
Comment (albert-magyar) [inline sim/Makefile:49]: I think the way this recipe is conservative but fast if unneeded is good.

File: sim/Makefile [hunk 2]
```diff
@@ -67,11 +77,11 @@ verilog: $(VERILOG)
 compile: $(VERILOG)
 
 # Phony targets for launching the sbt shell and running scalatests
-sbt: $(FIRRTL_JAR)
+sbt: $(FIRRTL_JAR) $(FIRRTL_TEST_JAR)
```
Comment (albert-magyar) [inline sim/Makefile:80]:
```suggestion
.PHONY: sbt
sbt: $(FIRRTL_JAR) $(FIRRTL_TEST_JAR)
```

File: sim/Makefile [hunk 3]
```diff
@@ -67,11 +77,11 @@ verilog: $(VERILOG)
...
 	cd $(base_dir) && $(SBT) "project $(firesim_sbt_project)" "shell"
-test: $(FIRRTL_JAR)
+test: $(FIRRTL_JAR) $(FIRRTL_TEST_JAR)
```
Comment (albert-magyar) [inline sim/Makefile:82]:
```suggestion
.PHONY: test
test: $(FIRRTL_JAR) $(FIRRTL_TEST_JAR)
```

File: sim/Makefile [hunk 4]
```diff
@@ -67,11 +77,11 @@ verilog: $(VERILOG)
...
 	cd $(base_dir) && $(SBT) "project $(firesim_sbt_project)" "test"
-testOnly: $(FIRRTL_JAR)
+testOnly: $(FIRRTL_JAR) $(FIRRTL_TEST_JAR)
```
Comment (albert-magyar) [inline sim/Makefile:84]:
```suggestion
.PHONY: testOnly
testOnly: $(FIRRTL_JAR) $(FIRRTL_TEST_JAR)
```

File: sim/src/main/makefrag/fasedtests/Makefrag [hunk 5]
```diff
@@ -40,11 +40,10 @@ endif
 submodules = \
     $(addprefix $(firesim_base_dir)/,. midas midas/targetutils firesim-lib) \
     $(addprefix $(chipyard_dir)/, \
-		$(addprefix generators/, rocket-chip rocket-chip/hardfloat rocket-chip/macros) \
-		$(addprefix tools/, firrtl chisel3 chisel3/chiselFrontend))
+		$(addprefix generators/, rocket-chip) \
```
Comment (albert-magyar) [inline sim/src/main/makefrag/fasedtests/Makefrag:43]: Is this an unrelated change?

File: sim/src/main/makefrag/midasexamples/Makefrag [hunk 6]
```diff
@@ -33,10 +33,9 @@ endif
 submodules = \
     $(addprefix $(firesim_base_dir)/,. midas midas/targetutils firesim-lib) \
     $(addprefix $(chipyard_dir)/, \
-		$(addprefix generators/, rocket-chip rocket-chip/hardfloat rocket-chip/macros) \
-		$(addprefix tools/, firrtl chisel3 chisel3/chiselFrontend))
+		$(addprefix generators/, rocket-chip) \
```
Comment (albert-magyar) [inline sim/src/main/makefrag/midasexamples/Makefrag:36]: Same here?

File: sim/src/main/makefrag/fasedtests/Makefrag (hunk 5 above)
Comment (davidbiancolin) [reply sim/src/main/makefrag/fasedtests/Makefrag:43]: Yeah, there's a separate commit for it.

File: sim/src/main/makefrag/midasexamples/Makefrag (hunk 6 above)
Comment (davidbiancolin) [reply sim/src/main/makefrag/midasexamples/Makefrag:36]: Ditto.

File: sim/Makefile (hunk 1 above)
Comment (davidbiancolin) [reply sim/Makefile:49]: Would it make sense to copy what firrtl does in build-scala?

**Comment by davidbiancolin** [comment]
It seems machine launch is failing to install verilator correctly in CI. I'll open a separate issue for that.
